cdc/sinkv2/eventsink/cloudstorage/dml_worker.go

```diff
@@ -45,15 +46,17 @@ type dmlWorker struct {
 	flushNotifyCh chan flushTask
 	// tableEvents maintains a mapping of <table, []eventFragment>.
 	tableEvents *tableEventsMap
-	// fileIndex maintains a mapping of <table, index number>.
-	fileIndex map[versionedTable]uint64
+	// fileIndex maintains a mapping of <table, indexWithDate>.
+	fileIndex map[versionedTable]*indexWithDate
 	// fileSize maintains a mapping of <table, file size>.
 	fileSize         map[versionedTable]uint64
 	wg               sync.WaitGroup
 	isClosed         uint64
 	errCh            chan<- error
 	extension        string
 	statistics       *metrics.Statistics
+	clock            clock.Clock
+	bufferPool       sync.Pool
```

Comment:
Is it necessary to benchmark the performance of the `bufferPool`.

Reply:
The reason why bufferPool is used is that a considerable number of `buffer.Grow-> runtime.growslice` are saw in heap profile. The benchmark will be added in later PR.